patches/minecraft/net/minecraft/block/Block.java.patch

```diff
@@ -207,6 +221,33 @@
 +    protected ThreadLocal<EntityPlayer> harvesters = new ThreadLocal();
 +    private ThreadLocal<IBlockState> silk_check_state = new ThreadLocal();
 +    protected static java.util.Random RANDOM = new java.util.Random(); // Useful for random things without a seed.
++
++    /**
++     * Called for ref-counting purposes by ExtendedBlockStorage in order to
++     * broadly cull a chunk from the random chunk update list for efficiency's sake.
++     *
++     * @param state The current block state
++     * @return Whether or not this block needs random ticking
++     */
++    public boolean getTickRandomly(IBlockState state)
++    {
++        return func_149653_t();
++    }
++
++    /**
++     * Called by WorldServer during an immediate block update in order to check
++     * whether or not new updates should occur on the same tick.
++     *
++     * @param world The current world
++     * @param pos Block position in world
++     * @param state The current block state
++     * @return Whether or not this block requires immediate updates
++     */
++    public boolean requiresUpdates(World world, BlockPos pos, IBlockState state)
```

Comment:
`requiresUpdates` isn't really representative of what the method does right?

I think it's okay give it a better name that doesn't match the MCP name, just edit the @Deprecated comment above so it's clear this one should be used instead.